File: packages/server-sdk/src/resources/projects.ts
import type { RavenHttpClient } from '../http-client';
import type { Project } from '../types';

/**
 * An API key is already permanently scoped to exactly one project (Phase 10
 * spec §2/§14), so there's nothing to list, create, update or delete here.
 * Those stay human, dashboard-session operations; not something a server
 * API key can or should do.
 *
 * `get()` takes no arguments. It always returns the one project this key
 * belongs to.
 */
export class ProjectsResource {
  constructor(private readonly http: RavenHttpClient) {}

  /** Returns the project this API key belongs to. */
  get(): Promise<Project> {
    return this.http.request<Project>('/v1/project');
  }
}

File: sdks/python/src/raven/resources/projects.py
"""An API key is already permanently scoped to exactly one project (Phase 10
spec Section 2/14) — there is nothing to list/create/update/delete here.
Those remain human/dashboard-session operations, not something a server API
key can or should do. ``get()`` takes no arguments: it always returns the one
project this key belongs to.
"""

from __future__ import annotations

from typing import cast

from .._async_http import AsyncRavenHttpClient
from .._http import RavenHttpClient
from .._types import Project


class ProjectsResource:
    def __init__(self, http: RavenHttpClient) -> None:
        self._http = http

    def get(self) -> Project:
        return cast(Project, self._http.request("/v1/project"))


class AsyncProjectsResource:
    def __init__(self, http: AsyncRavenHttpClient) -> None:
        self._http = http

    async def get(self) -> Project:
        return cast(Project, await self._http.request("/v1/project"))
